src/shims/fs.rs

```diff
@@ -44,7 +43,8 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         let mut options = OpenOptions::new();
 
-        // The first two bits of the flag correspond to the access mode of the file in linux.
+        // The first two bits of the flag correspond to the access mode of the file in linux. This
+        // is done this way because `O_RDONLY` is zero in several platforms.
         let access_mode = flag & 0b11;
```

Comment:
Is this `0b11` backed by anything? Like, how to you know that the access flags are in the lowest two bits on all POSIX platforms?

Reply:
We don't. I tried to do this as platform agnostic as possible but the main issue is that the read only flag is `0` in linux, windows and mac iirc. So doing `flags & O_RDONLY != 0` as with the other flags won't work. 

The `0b11` comes from the fact that in our 3 supported platforms the read only, write only and read write flags are `0`, `1` and `2` respectively.

Reply:
Hm, makes you wonder how they parse their flags then...

Seems fine not to support other platforms but we should assert that. Something `O_RDONLY | O_WRONLY | O_RDWR & ~0b11 == 0` or so.

Reply:
I think they just have a bunch of conditionals for every other flag and then they assume not being in any of those means that they are in `O_RDONLY`... Who knows? Also `0b11` as a flag doesn't have a defined behavior ¬¬

Anyway. Do you want me to literally assert this and panic? or just throw an unsupported error with `"Cannot parse access mode flags"`?

Reply:
Unsupported error makes more sense indeed, with a message like "access mode flags on this platform are unsupported".